src/pages/setpassword/SetPasswordForm/SetPasswordFormNarrow.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import {
+    Image,
+    Text, TextInput, View,
+} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'underscore';
+import lodashGet from 'lodash.get';
+import styles from '../../../styles/styles';
+import ButtonWithLoader from '../../../components/ButtonWithLoader';
+import {setPassword} from '../../../libs/actions/Session';
+import ONYXKEYS from '../../../ONYXKEYS';
+import SetPasswordPageProps from '../SetPasswordPageProps';
+import welcomeScreenshot from '../../../../assets/images/welcome-screenshot.png';
+
+class SetPasswordFormNarrow extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.submitForm = this.submitForm.bind(this);
+
+        this.state = {
+            password: '',
+            formError: null,
+        };
+    }
+
+    /**
+     * Validate the form and then submit it
+     */
+    submitForm() {
+        if (!this.state.password.trim()) {
+            this.setState({
+                formError: 'Password cannot be blank',
+            });
+            return;
+        }
+
+        this.setState({
+            formError: null,
+        });
+        setPassword(this.state.password, lodashGet(this.props.route, 'params.validateCode', ''));
+    }
```

Comment:
Can we apply the DRY principle here somehow? These two components both use the same `submitForm()` method.